File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/adapter/TokenRepositoryAdapter.java
package spring.web.java.infrastructure.persistence.adapter;

import java.util.Optional;

import org.springframework.stereotype.Repository;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.model.entity.Token;
import spring.web.java.domain.repository.TokenRepository;
import spring.web.java.infrastructure.persistence.TokenRedisRepository;

@Repository
@RequiredArgsConstructor
public class TokenRepositoryAdapter implements TokenRepository {

	private final TokenRedisRepository redisRepository;

	@Override
	public Optional<Token> findById(Long id) {
		return redisRepository.findById(id);
	}

	@Override
	public Token save(Token token) {
		return redisRepository.save(token);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/TokenRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Component
import spring.web.kotlin.domain.model.entity.Token
import spring.web.kotlin.domain.repository.TokenRepository
import spring.web.kotlin.infrastructure.persistence.TokenRedisRepository

@Component
class TokenRepositoryAdapter(
    private val redisRepository: TokenRedisRepository
) : TokenRepository {
    override fun findByIdOrNull(id: Long) = redisRepository.findByIdOrNull(id)

    override fun save(token: Token) = redisRepository.save(token)
}